src/main/java/com/impactupgrade/nucleus/util/PageResultsProcessor.java

```diff
@@ -0,0 +1,31 @@
+package com.impactupgrade.nucleus.util;
+
+import com.google.common.base.Strings;
+import com.impactupgrade.nucleus.model.PagedResults;
+
+import java.util.function.Consumer;
+
+public class PageResultsProcessor<T> {
```

Comment:
Once the below is tweaked (if you agree), can you work this into other areas that need to leverage this? They're not paging at all currently and should be...

ConstantContactCommunicationService.syncContacts
MinistryByTextCommunicationService.syncContacts
SendGridCommunicationService.syncContacts
StripeDataSyncService.syncContacts
XeroDataSyncService.syncContacts